src/hooks/screen/useCreateChat.ts

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import SimpleToast from 'react-native-simple-toast';
+import {v4 as uuid} from 'uuid';
+
+import {getOrCreateAnonymousChannel} from '../../service/chat';
+import SignedMessageRepo from '../../service/repo/signedMessageRepo';
+import ChannelList from '../../database/schema/ChannelListSchema';
+import useLocalDatabaseHook from '../../database/hooks/useLocalDatabaseHook';
+import UserSchema from '../../database/schema/UserSchema';
+import ChannelListMemberSchema from '../../database/schema/ChannelListMemberSchema';
+import useChatUtilsHook from '../core/chat/useChatUtilsHook';
+import {GROUP_INFO} from '../core/constant';
+
+const useCreateChat = () => {
+  const [loadingCreateChat, setLoadingCreateChat] = React.useState(false);
+  const {localDb} = useLocalDatabaseHook();
+  const {goToChatScreen} = useChatUtilsHook();
+  const createSignChat = async (members: string[], selectedUser, from) => {
+    try {
+      setLoadingCreateChat(true);
+      const initChannel = await SignedMessageRepo.createSignedChat(members);
+      const chatData = await createChannelJson(initChannel, selectedUser);
+
+      const channelList = await ChannelList.fromMessageSignedAPI(chatData);
+      channelList.saveIfLatest(localDb);
+      handleMemberSchema(initChannel);
+      setLoadingCreateChat(false);
+      goToChatScreen(channelList, from);
+    } catch (e) {
+      setLoadingCreateChat(false);
+      console.log({e}, 'error create chat');
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `createSignChat` function is missing error handling for the case when `SignedMessageRepo.createSignedChat(members)` or `ChannelList.fromMessageSignedAPI(chatData)` fails. If these functions throw an error, it will be caught by the catch block and the loading state will be set to false, but the user won't be notified about the failure. Consider adding a toast notification in the catch block to inform the user about the error.

```diff
    } catch (e) {
      setLoadingCreateChat(false);
+      SimpleToast.show('Failed to create chat');
      console.log({e}, 'error create chat');
    }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->